react/features/remote-video-menu/components/web/MuteEveryoneDialog.js

```diff
@@ -0,0 +1,137 @@
+/* @flow */
+
+import React from 'react';
+
+import {
+    AUDIO_MUTE,
+    createToolbarEvent,
+    sendAnalytics
+} from '../../../analytics';
+import { Dialog } from '../../../base/dialog';
+import { translate } from '../../../base/i18n';
+import { connect } from '../../../base/redux';
+import { setAudioMuted } from '../../../base/media';
+import UIEvents from '../../../../../service/UI/UIEvents';
+
+import AbstractMuteRemoteParticipantDialog, {
+    type Props as AbstractProps
+} from '../AbstractMuteRemoteParticipantDialog';
+
+declare var APP: Object;
+
+/**
+ * The type of the React {@code Component} props of
+ * {@link MuteEveryoneDialog}.
+ */
+type Props = AbstractProps & {
+
+    /**
+     * The ID of the local participant.
+     */
+    localParticipantId: string,
+
+    /**
+     * The IDs of the remote participants to be muted.
+     */
+    remoteParticipantIds: Array<string>,
+
+     /**
+     * The IDs of the remote participants to exclude from being muted.
+     */
+    exclude: Array<string>,
+};
+
+/**
+ * A React Component with the contents for a dialog that asks for confirmation
+ * from the user before muting a remote participant.
+ *
+ * @extends Component
+ */
+class MuteEveryoneDialog extends AbstractMuteRemoteParticipantDialog<Props> {
+    static defaultProps = {
+        exclude: [],
+        muteLocal: false
+    };
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const { t } = this.props;
+
+        return (
+            <Dialog
+                okKey = 'dialog.muteMultipleParticipantButton'
+                onSubmit = { this._onSubmit }
+                titleKey = 'dialog.muteMultipleParticipantTitle'
+                width = 'small'>
+                <div>
+                    { t('dialog.muteMultipleParticipantBody') }
+                </div>
+            </Dialog>
+        );
+    }
+
+    _onSubmit: () => boolean;
+
+    /**
+     * Callback to be invoked when the value of this dialog is submitted.
+     *
+     * @returns {boolean}
+     */
+    _onSubmit() {
+        const {
+            dispatch,
+            exclude,
+            localParticipantId,
+            remoteParticipantIds
+        } = this.props;
+
+        remoteParticipantIds
```

Comment:
Let's move the whole mute logic to a separate action. It will probably need to be async action to select the participants.

This way you won't need to `mapStateToProps` at all and the dialog wouldn't be rerendered every time when something changes in the participants.

Reply:
Good point - updated.